usr/lib/zabbix-check/bin/zabbix-check-spamassassin

```diff
@@ -1,21 +1,22 @@
 #!/bin/bash
 
 if [ -S /var/run/spamd.sock ]; then
-        timeout 5 socat - UNIX-CONNECT:/var/run/spamd.sock <<< "PING SPAMC/1.0" | grep -q PONG
-	echo $?
-	exit 0
+    timeout 3 socat - UNIX-CONNECT:/var/run/spamd.sock <<< "PING SPAMC/1.0" | grep -q PONG
+    echo $?
+    exit 0
 fi
 
-if [ "$(netstat -ltn | grep 783 | wc -l)" -gt "0" ]; then 
-	/etc/init.d/spamassassin status >/dev/null 2>&1
-	r=$?
-	if [ $r -eq 0 ]; then
-		echo 1
-	elif [ $r -eq 1 ]; then
-		echo 0
-	else
-		echo $r
-	fi
-else
-	echo 0
+if [ "$(netstat -ltn | grep 783 | wc -l)" -gt "0" ]; then
+    /etc/init.d/spamassassin status >/dev/null 2>&1
```

Comment:
Is this working in a systemd environment?

Reply:
yes, systemd returns a status code in ubuntu

Reply:
Ah the sysvinit script calls systemd, I see.